SofaKernel/modules/SofaCore/src/sofa/core/behavior/PairStateAccessor.h

```diff
@@ -0,0 +1,100 @@
+﻿/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/core/behavior/StateAccessor.h>
+#include <sofa/core/behavior/MechanicalState.h>
+
+namespace sofa::core::behavior
+{
+
+/**
+ * Base class for components having access to a pair of mechanical states with a specific template parameter, in order
+ * to read and/or write state variables.
+ */
+template<class DataTypes1, class DataTypes2 = DataTypes1>
+class PairStateAccessor : public virtual StateAccessor
+{
+public:
+    SOFA_CLASS(SOFA_TEMPLATE2(PairStateAccessor, DataTypes1, DataTypes2), StateAccessor);
+
+    void init() override;
+
+    /// Retrieve the associated MechanicalState #1
+    MechanicalState<DataTypes1>* getMState1() { return mstate1; }
```

Comment:
Maybe a const alternative here could be useful?

```suggestion
    MechanicalState<DataTypes1>* getMState1() { return mstate1; }
    const MechanicalState<DataTypes1>* getMState1() const { return mstate1; }
```

and similarly for other accessors.

Reply:
You are right. I just copy/pasted the existing functions